src/main/java/games/strategy/thread/ThreadPool.java

```diff
@@ -53,11 +53,7 @@ public void waitForAll() {
       } catch (final InterruptedException e) {
         Thread.currentThread().interrupt();
       } catch (final ExecutionException e) {
-        // ExecutionException contains no useful information; it's simply an adapter to tunnel
-        // exceptions thrown by tasks through the Executor API. Log the cause only to reducase
-        // stack trace frames.
-        ClientLogger.logError("Threading execution exception: " + e.getCause().getMessage(),
-            e.getCause());
+        ClientLogger.logError("Threading execution exception: " + e.getMessage(), e);
```

Comment:
`e.getMessage()` and `e.getCause().getMessage()` should always yield the same value when `e` is an instance of `ExecutionException`.  I searched JDK 8 for uses of `ExecutionException(String)` and `ExecutionException(String, Throwable)` and didn't find any results, which means no custom message is ever specified that would override the cause's message.